TestingLowerBounds/Testing/Binary.lean

```diff
@@ -166,25 +185,89 @@ instance (π : Measure Bool) [IsFiniteMeasure π] : IsMarkovKernel (twoHypKernel
     simp [h]
   · infer_instance
 
+--the finiteness hypothesis for μ should not be needed, but otherwise I dont know how to handle the 3rd case, where I have the complement
+-- we still need some hp, the right one is probably SigmaFinite. For SFinite ones there is a counterexample, see the comment above `Measure.prod_eq`.
+--TODO: generalize this lemma to SigmaFinite measures, there are 2 ways to do it, one is to try and generalize this proof (for this it may be useful to try and apply the lemma used in the proof of `Measure.prod_eq`), the other is to use this as an auxiliary lemma and prove the result for SigmaFinite measures using this one (we can restrict the mesaure to the set where it is finite and then use this lemma). I'm not sure which way is better.
+lemma measure_prod_ext {μ ν : Measure (𝒳 × 𝒴)} [IsFiniteMeasure μ] (h : ∀ (A : Set 𝒳),
+    ∀ (_ : MeasurableSet A), ∀ (B : Set 𝒴), ∀ (_ : MeasurableSet B), (μ (A ×ˢ B)) = (ν (A ×ˢ B))) :
+    μ = ν := by
+  ext s hs
+  apply MeasurableSpace.induction_on_inter generateFrom_prod.symm isPiSystem_prod _ _ _ _ hs
+  · simp_rw [measure_empty]
+  · exact fun t ⟨A, hA, B, hB, ht⟩ ↦ ht ▸ h A hA B hB
+  · intro t ht h_eq
+    rw [measure_compl ht (measure_ne_top μ t), measure_compl ht (h_eq ▸ measure_ne_top μ t), h_eq,
+      ← Set.univ_prod_univ, ← h Set.univ MeasurableSet.univ Set.univ MeasurableSet.univ]
+  · intro A h_disj h_meas h_eq
+    simp_rw [measure_iUnion h_disj h_meas, h_eq]
+
+--put these 2 lemmas in a separate file, maybe PR them to mathlib
+lemma _root_.Prod.swap_image {α β : Type*} (A : Set α) (B : Set β) :
+    Prod.swap '' (A ×ˢ B) = B ×ˢ A := by ext; simp
```

Comment:
This is [Set.image_swap_prod](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Set/Prod.html#Set.image_swap_prod)